Core/Consumers/AbstractAsyncConsumer.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Smartbox\Integration\FrameworkBundle\Core\Consumers;
+
+use PhpAmqpLib\Message\AMQPMessage;
+use Smartbox\CoreBundle\Utils\Helper\DateTimeCreator;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesLogger;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
+use Smartbox\Integration\FrameworkBundle\Events\TimingEvent;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class AbstractConsumer.
+ */
+abstract class AbstractAsyncConsumer extends Service implements ConsumerInterface
+{
+    use IsStopableConsumer;
+    use UsesLogger;
+    use UsesSmartesbHelper;
+
+    /**
+     * Sleep flag. Prevents the consumer from running too fast and causing a CPU usage spike when there are
+     * no messages available.
+     *
+     * @var bool
+     */
+    protected $sleep = true;
+
+    /**
+     * Holds the amount of time it took to process a message once the callback is triggered. Due to the asynchronicity
+     * nature of this consumer, it's not possible to measure how long it takes to get a message from the queue and send
+     * it to the callback
+     *
+     * @var int
+     */
+    protected $consumptionDuration = 0;
+
+    /**
+     * Initializes the consumer for a given endpoint.
+     *
+     * @param EndpointInterface $endpoint
+     */
+    abstract protected function initialize(EndpointInterface $endpoint);
+
+    /**
+     * Cleans up and closes connections before shutting down.
+     *
+     * @param EndpointInterface $endpoint
+     *
+     * @return mixed
+     */
+    abstract protected function cleanUp(EndpointInterface $endpoint);
+
+    /**
+     * After the execution of this method, it will be considered that the message was successfully handled,
+     * therefore, if there was any problem, an exception must be thrown and not continue. This is important to ensure
+     * the Message Delivery Guarantee.
+     *
+     * @param MessageInterface $message
+     */
+    protected function process(EndpointInterface $endpoint, QueueMessageInterface $message)
+    {
+        $endpoint->handle($message);
+    }
+
+    /**
+     * This function is called to confirm that a message was successfully handled. Until this point, the message should
+     * not be removed from the source Endpoint, this is very important to ensure the Message delivery guarantee.
+     *
+     * @param EndpointInterface $endpoint
+     * @param $message
+     *
+     * @return MessageInterface
+     */
+    abstract protected function confirmMessage(EndpointInterface $endpoint, QueueMessageInterface $message);
```

Comment:
This abstract consumer is not related to queues, so this means that we can't use the QueueMessageInterface at that level.

Reply:
Fixed

Reply:
The same here, it is still not fixed